crates/matrix-sdk/src/sync.rs

```diff
@@ -154,6 +154,19 @@ impl Client {
     ) -> Result<()> {
         let BaseSyncResponse { rooms, presence, account_data, to_device, notifications } = response;
 
+        {
```

Comment:
I feel like this is a bit of the wrong place for this? Perhaps a separate method would be in place, something like `update_cached_values()` which we're doing to call before calling this handlers method.

Reply:
You're right, I've spotted the first common place to both syncs, but this should rather be somewhere in the base crate, for it to make sense.